src/app/reporting/Engine.cpp

```diff
@@ -636,7 +636,8 @@ void Engine::Run()
         ReadHandler * readHandler = imEngine->ActiveHandlerAt(mCurReadHandlerIdx % (uint32_t) imEngine->mReadHandlers.Allocated());
         VerifyOrDie(readHandler != nullptr);
 
-        if (imEngine->GetReportScheduler()->IsReportableNow(readHandler))
+        if (readHandler->IsType(ReadHandler::InteractionType::Read) || readHandler->IsPriming() ||
```

Comment:
It might be better to have a MustReportIfPossible() (or some such name; I am having a hard time thinking of one right now) on readHandler which exactly returns `IsType(ReadHandler::InteractionType::Read) || IsPriming()`.  We could use that here, and in a few places above....  And it would avoid this dependency on details about priming and whatnot here.

Reply:
How about CanEmitReadReport() ?

Reply:
But it applies to subscriptions too.

Also, I just realized the logic here should really be checking for `CanStartReporting() or equivalent` as well: we should not be triggering reports here for a ReadHandler that is not at `CanStartReporting` state yet, right?

Maybe another way to approach this: a read handler can be in one of two situations.  It could just be reportable, no matter what time it is (this is the read or priming case), or its reportability might depend on time (subscription steady-state).  The problem is that IsReportableAtAllTimes is long and kind of still misleading/confusing...  

What do you think of this approach?

```
// CanStartReporting() is true if the ReadHandler is in a state where it could generate
// a (possibly empty) report if someone asked it to.
bool CanStartReporting() const { return mState == HandlerState::CanStartReporting; }
// ShouldReportUnscheduled() is true if the ReadHandler should be asked to generate reports
// without consulting the report scheduler.
bool ShouldReportUnscheduled() const { return CanStartReporting() && (IsType(ReadHandler::InteractionType::Read) || IsPriming()); }
```

`ShouldReportUnscheduled` could be called from `MoveToState`.  It would redundantly check the state, but that's probably OK....  And it could be called from `ShouldStartReporting` too....

I considered calling it `IsReportableNow` instead of `ShouldReportUnscheduled, but the possible confusion between that name and the IsReportableNow on the scheduler was bothering me....  This makes it a bit clearer that this ReadClient is in a mode where the scheduler is not managing it.

What about `ShouldReportUnscheduled()`?

Reply:
Changed to ShouldReportUnscheduled() and added a check to !IsActiveSubscription() to ensure the scheduler isn't skipped in the MoveToState at the end of OnStatusResponse (Right before we remove the IsPriming flag).